mixer/pkg/lang/cel/provider.go

```diff
@@ -249,7 +251,7 @@ func (v value) IsSet(index ref.Val) ref.Val {
 	return types.True
 }
 
-func (a attributeActivation) ResolveName(name string) (ref.Val, bool) {
+func (a attributeActivation) ResolveName(name string) (interface{}, bool) {
 	if node, ok := a.provider.root.children[name]; ok {
 		return resolve(node, a.bag), true
```

Comment:
Note, the resolve call does type adaptation to a CEL value under the covers and returns a `ref.Val`. If you want to take advantage of speed improvements related to attribute resolution you might want to port this logic over to an `interpreter.CustomAttributeFactory`. This change could be done in a future PR.